pallets/loans/src/lib.rs

```diff
@@ -1208,7 +1236,8 @@ pub mod pallet {
 		}
 
 		fn update_nav(pool_id: T::PoolId) -> Result<T::Balance, DispatchError> {
-			Ok(Self::update_portfolio_valuation_for_pool(pool_id)?.0)
+			Self::update_portfolio_valuation_for_pool(pool_id, PriceCollectionInput::FromRegistry)
```

Comment:
A quick comprehension question: Our rather newer NAV calculation runtime API calls `Loans::update_nav(pool_id)` such that the Pool Fees NAV calculation can be based on that. Say the loans NAV is updated with a price collection input in block `i` and in the next block `i+1` we are querying the NAV via `PoolsApi::nav(pool_id)` which performs `Loans::update_nav(pool_id)` with an update based on `PriceCollectionInput::FromRegistry`. IIUC, the prices from the update in block `i` are still cached. What I want to  find out is if `Loans::update_nav` can make the Loans NAV less accurate if it was previously based on a custom price list.

Reply:
It was previously based on the price registry, so both lines above are exactly identical. From a pool fees perspective, nothing has changed. The only open door (right now) is for the RuntimeAPI.